presentation/common/src/presentation-common/RulesetVariables.ts

```diff
@@ -36,30 +36,154 @@ export enum VariableValueTypes {
+  type: VariableValueTypes.IntArray;
+  value: number[];
+}
+/**
+ * Data structure for representing ID ruleset variables.
+ * @public
+ */
+export interface Id64RulesetVariable extends RulesetVariableBase {
+  type: VariableValueTypes.Id64;
+  value: Id64String;
+}
+/**
+ * Data structure for representing ID array ruleset variables.
+ * @public
+ */
+export interface Id64sRulesetVariable extends RulesetVariableBase {
+  type: VariableValueTypes.Id64Array;
+  value: Id64String[];
+}
+/**
+ * Data structure for representing ruleset variables.
+ * @public
+ */
+export type RulesetVariable = BooleanRulesetVariable | StringRulesetVariable | IntRulesetVariable | IntsRulesetVariable | Id64RulesetVariable | Id64sRulesetVariable;
 
 /**
- * JSON representation of [[RulesetVariable]].
+ * JSON representation of [[RulesetVariableBase]].
  * @public
  */
-export interface RulesetVariableJSON {
+export interface RulesetVariableBaseJSON {
   id: string;
   type: VariableValueTypes;
   value: VariableValueJSON;
 }
+/**
+ * JSON representation of [[BooleanRulesetVariable]].
+ * @public
+ */
+export interface BooleanRulesetVariableJSON extends RulesetVariableBaseJSON {
+  type: VariableValueTypes.Bool;
+  value: boolean;
+}
+/**
+ * JSON representation of [[StringRulesetVariable]].
+ * @public
+ */
+export interface StringRulesetVariableJSON extends RulesetVariableBaseJSON {
+  type: VariableValueTypes.String;
+  value: string;
+}
+/**
+ * JSON representation of [[IntRulesetVariable]].
+ * @public
+ */
+export interface IntRulesetVariableJSON extends RulesetVariableBaseJSON {
+  type: VariableValueTypes.Int;
+  value: number;
+}
+/**
+ * JSON representation of [[IntsRulesetVariable]].
+ * @public
+ */
+export interface IntsRulesetVariableJSON extends RulesetVariableBaseJSON {
+  type: VariableValueTypes.IntArray;
+  value: number[];
+}
+/**
+ * JSON representation of [[Id64RulesetVariable]].
+ * @public
+ */
+export interface Id64RulesetVariableJSON extends RulesetVariableBaseJSON {
+  type: VariableValueTypes.Id64;
+  value: Id64String;
+}
+/**
+ * JSON representation of [[Id64sRulesetVariable]].
+ * @public
+ */
+export interface Id64sRulesetVariableJSON extends RulesetVariableBaseJSON {
+  type: VariableValueTypes.Id64Array;
+  value: CompressedId64Set;
+}
+/**
+ * JSON representation of [[RulesetVariable]].
+ * @public
+ */
+export type RulesetVariableJSON = BooleanRulesetVariableJSON | StringRulesetVariableJSON | IntRulesetVariableJSON | IntsRulesetVariableJSON | Id64RulesetVariableJSON | Id64sRulesetVariableJSON;
+
+/** @public */
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export namespace RulesetVariable {
+  /** Serialize given RulesetVariable to JSON */
+  export function toJSON(variable: RulesetVariable): RulesetVariableJSON {
+    if (variable.type === VariableValueTypes.Id64Array)
+      return { ...variable, value: CompressedId64Set.compressArray(variable.value) };
```

Comment:
Do you know that variable.value is always already sorted according to the criterion required by `CompressedId64Set.compressArray`?

Reply:
When `Id64String[]` value is passed to us on the frontend, I sort them using `OrderedId64Iterable.sortArray` and store sorted. 

But now that you mentioned this, I remembered there's another way to pass these variables in which case I'm not sorting them - I'll fix that. Thanks